### DISCUSSION TITLE
Difference when using two grid definitions with media query

### DISCUSSION THREAD

**Discussion by afucher** [tailwindlabs/tailwindcss#9581] [Help] [answered]
I created a footer that in smaller screens it uses a 3 rows grid, and in larger screens it uses a 3 cols grid. To achieve that I'm using this classes `grid grid-rows-3 justify-items-center sm:grid-cols-3`
In smaller screens it is working as expected, but in larger screens it is creating the div with a unecessary height. If I remove and use the cols-3 by default it not create this unecessary space in the bottom of the screen: `grid justify-items-center grid-cols-3`

Using Tailwind Play I'm able to reproduce, those are the differences:
**With media queries to change between cols and rows**
<img width="1274" alt="Captura de Tela 2022-10-16 às 17 57 55" src="https://user-images.githubusercontent.com/3756185/196058005-2bfd605f-5b25-48d8-baee-812d04b17704.png">
**Using by default the cols config**
<img width="1270" alt="Captura de Tela 2022-10-16 às 17 58 21" src="https://user-images.githubusercontent.com/3756185/196058017-b9a888e2-e645-442e-8237-4681cc9fd0c0.png">

This is the code snippet to reproduce:
```
<main class="">
  <footer class="max-w-7xl bg-blue-900 px-4 pt-10 text-blue-100">
    <div class="">
      <div class="grid grid-rows-3 justify-items-center sm:grid-cols-3">
        <div>
          <p>Address 1</p>
          <p>Address 2</p>
          <p>Address 3</p>
          <p>Address 4</p>
        </div>
        <div>
          <ul>
            <li>Instagram</li>
            <li>Twitter</li>
            <li>Facebook</li>
            <li>YouTube</li>
          </ul>
        </div>
        <div class="grid grid-cols-2">
          <div>
            <ul>
              <li>Cursos</li>
              <li>Formação</li>
              <li>Sobre</li>
              <li>Contato</li>
            </ul>
          </div>
          <div>
            <ul>
              <li>Software</li>
              <li>Professores</li>
              <li>Serviços</li>
              <li>Acesso aluno</li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </footer>
</main>
```

Maybe is some Tailwind or CSS knowledge that I don't have, can you please help me how to avoid this unecessary height?

**Comment by wongjn** [comment, marked as answer]
`grid-rows-3` means `grid-template-rows: repeat(3, 1fr)` where <q>the `fr` unit represents a fraction of the leftover space in the grid container.</q>[^1] So for `repeat(3, 1fr)`, each `fr` means a 3rd. When all items are in one row when the `sm:grid-cols-3` applies, this means there are still three rows, each with the same height specified by the first row content, thus the empty space.

To fix your issue, you can reduce the row count to 1 when it switches to the 3-column layout with `sm:grid-rows-1`.

[^1]: https://developer.mozilla.org/en-US/docs/Web/CSS/flex_value#syntax

**Reply by afucher** [reply]
Thanks! Worked.. changed to `grid grid-rows-3 sm:grid-rows-1 sm:grid-cols-3`
